src/transform/src/map_lifting.rs

```diff
@@ -57,10 +59,9 @@ impl LiteralLifting {
     /// expressions as it goes.
     ///
     /// In several cases, we only manage to extract literals from the final
-    /// columns. This could be improved using permutations to move all of
-    /// the literals to the final columns, and then rely on projection
-    /// hoisting to allow the these literals to move up the AST.
-    // TODO(frank): Fix this.
+    /// columns. But in those cases where it is possible, permutations are
+    /// used to move all of the literals to the final columns, and then rely
+    /// on projection hoisting to allow the these literals to move up the AST.
```

Comment:
Can you please add a comment here about how final literals are currently treated differently from not-final literals? I.e. currently this transform will essentially turn something like:
```
Constant [1 2 3] [1 4 3]
```
into
```
Constant [2] [4]
Map 1
Project [1, 0]
Map 3
```

Also, can you create a github issue for the work to required to make it unnecessary to treat final and non-final literals differently? Then reference that issue in a TODO here?

Reply:
Yes, sorry, you told me that in our 1:1. I had it in my list, but got stuck in my other task.

Reply:
Logged https://github.com/MaterializeInc/materialize/issues/6598.